### PR TITLE
change margin to account for more than one anchor

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adds `margin` to the `right` side of the `anchor` in a banner notification
#### Where should the reviewer start?
Notification.js
#### How should this be manually tested?
put more than one `anchor` 
```
            actions={[
              {
                href: '#',
                label: 'Renew Subscription',
              },
              {
                href: '#',
                label: 'Renew',
              },
              {
                href: '#',
                label: 'Renew',
              },
            ]}

```

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
previously there was only `margin` on the left side which when putting more than one `anchor` there was no space between them 
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Notification/Notification.js [hunk 1]
```diff
@@ -137,9 +137,11 @@ const Notification = ({
     actions = actionsProp.map((action, index) => (
       <NotificationAnchor
         key={action.label}
-        // create space between first anchor and text content
+        // create space between first anchor and text content and next anchor
         margin={
-          index === 0 && (message || title) ? { left: 'xsmall' } : undefined
+          index >= 0 && (message || title)
```
Comment (taysea) [inline src/js/components/Notification/Notification.js:142]: seems like we want the margin to apply to all anchors, and the first anchor if there is a message or a title.

```
let margin;
if (index === 0 && (message || title)) margin = { left: 'xsmall' };
else if (index > 0) margin = { left: 'xsmall' };

...
 margin={margin}

File: src/js/components/Notification/Notification.js [hunk 2]
```diff
@@ -134,16 +137,19 @@ const Notification = ({
...
-        margin={
-          index === 0 && (message || title) ? { left: 'xsmall' } : undefined
-        }
-        {...action}
-      />
-    ));
+    actions = actionsProp.map((action, index) => {
+      let margin;
+      if (index === 0 && (message || title)) margin = { left: 'xsmall' };
+      else if (index > 0) margin = { left: 'xsmall' };
```
Comment (taysea) [inline src/js/components/Notification/Notification.js:143]: Can we add two anchors in the 'actions' Notifications test to make sure this condition is captured in the snapshot?

Comment (britt6612) [reply src/js/components/Notification/Notification.js:143]: yes

File: src/js/components/Notification/Notification.js [hunk 3]
```diff
@@ -139,23 +139,25 @@ const Notification = ({
   if (actionsProp)
     actions = actionsProp.map((action, index) => {
       let margin;
-      if (index === 0 && (message || title)) margin = { left: 'xsmall' };
-      else if (index > 0) margin = { left: 'xsmall' };
+      if (index === 0 && (message || title)) margin = { right: 'xsmall' };
```
[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/Notification/Notification.js:142]: so even if we do not have any anchors we still put `margin` would that be fine or just put `margin` for the case there are anchors

File: src/js/components/Notification/Notification.js [hunk 4]
```diff
@@ -139,23 +139,25 @@ const Notification = ({
...
-          margin={margin}
-          {...action}
-        />
+        <Fragment key={action.label}>
+          <NotificationAnchor
+            // create space between first anchor and
+            // text content and next anchor
+            margin={margin}
+            {...action}
+          />{' '}
```
[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/Notification/Notification.js:151]: yes I will add

File: src/js/components/Notification/Notification.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Notification/Notification.js:142]: In this scenario, we are putting the margin on the anchors. So, if we don't have any anchors, we won't put any margin on them.

[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/Notification/Notification.js:142]: good point sorry end of day!

**Review by ericsoderberghp — APPROVED** [review]

**Comment by britt6612** [comment]
I just added a way to be able to control the `anchor` color through the theme

**Review by taysea — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**